packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -0,0 +1,1136 @@
+  /// than maxScale.
+  final double minScale;
+
+  /// Called when the user ends a pan or scale gesture on the widget.
+  ///
+  /// {@template flutter.widgets.interactiveViewer.onInteraction}
+  /// Will be called even if the interaction is disabled with
+  /// [translationEnabled] or [scaleEnabled].
+  ///
+  /// A [GestureDetector] wrapping the InteractiveViewer will not respond to
+  /// [GestureDetector.onScaleStart], [GestureDetector.onScaleUpdate], and
+  /// [GestureDetector.onScaleEnd]. Use [onInteractionStart],
+  /// [onInteractionUpdate], and [onInteractionEnd] to respond to those
+  /// gestures.
+  ///
+  /// The coordinates returned in the details are viewport coordinates relative
+  /// to the parent. See [TransformationController.toScene] for how to
+  /// convert the coordinates to scene coordinates relative to the child.
+  /// {@endtemplate}
+  ///
+  /// See also:
+  ///
+  ///  * [onInteractionStart], which handles the start of the same interaction.
+  ///  * [onInteractionUpdate], which handles an update to the same interaction.
+  final GestureScaleEndCallback onInteractionEnd;
+
+  /// Called when the user begins a pan or scale gesture on the widget.
+  ///
+  /// {@macro flutter.widgets.interactiveViewer.onInteraction}
+  ///
+  /// See also:
+  ///
+  ///  * [onInteractionUpdate], which handles an update to the same interaction.
+  ///  * [onInteractionEnd], which handles the end of the same interaction.
+  final GestureScaleStartCallback onInteractionStart;
+
+  /// Called when the user updates a pan or scale gesture on the widget.
+  ///
+  /// {@macro flutter.widgets.interactiveViewer.onInteraction}
+  ///
+  /// See also:
+  ///
+  ///  * [onInteractionStart], which handles the start of the same interaction.
+  ///  * [onInteractionEnd], which handles the end of the same interaction.
+  final GestureScaleUpdateCallback onInteractionUpdate;
+
+  /// A [TransformationController] for the transformation performed on the
+  /// child.
+  ///
+  /// Whenever the child is transformed, the [Matrix4] value is updated and all
+  /// listeners are notified. If the value is set, InteractiveViewer will update
+  /// to respect the new value.
+  ///
+  /// {@tool dartpad --template=stateful_widget_material_ticker}
+  /// This example shows how transformationController can be used to animate the
+  /// transformation back to its starting position.
+  ///
+  /// ```dart
+  /// final TransformationController _transformationController = TransformationController();
+  /// Animation<Matrix4> _animationReset;
+  /// AnimationController _controllerReset;
+  ///
+  /// void _onAnimateReset() {
+  ///   _transformationController.value = _animationReset.value;
+  ///   if (!_controllerReset.isAnimating) {
+  ///     _animationReset?.removeListener(_onAnimateReset);
+  ///     _animationReset = null;
+  ///     _controllerReset.reset();
+  ///   }
+  /// }
+  ///
+  /// void _animateResetInitialize() {
+  ///   setState(() {
```

Comment:
It looks like you don't really need this setState?

Reply:
That's right, good catch.